cli/utils/start-e2e.js

```diff
@@ -0,0 +1,207 @@
+  logger.info(`Exiting process with ${exitCode}`);
+  process.exit(exitCode || 0);
+}
+
+/**
+ * Spawns the protractor command.
+ * Perhaps this should be API driven?
+ * @name spawnProtractor
+ */
+function spawnProtractor(configPath, chunks, port, skyPagesConfig) {
+  logger.info('Running Protractor');
+  protractorLauncher.init(configPath, {
+    params: {
+      localUrl: `https://localhost:${port}`,
+      chunks: chunks,
+      skyPagesConfig: skyPagesConfig
+    }
+  });
+  process.on('exit', killServers);
+}
+
+/**
+ * Spawns the selenium server if directConnect is not enabled.
+ * @name spawnSelenium
+ */
+function spawnSelenium(configPath) {
+  const config = require(configPath).config;
+
+  return new Promise((resolve, reject) => {
+    logger.info('Spawning selenium...');
+
+    // Assumes we're running selenium ourselves, so we should prep it
+    if (config.seleniumAddress) {
+      logger.info('Installing Selenium...');
+      selenium.install({ logger: logger.info }, () => {
+        logger.info('Selenium installed. Starting...');
+        selenium.start((err, child) => {
+          if (err) {
+            reject(err);
+            return;
+          }
+
+          seleniumServer = child;
+          logger.info('Selenium server is ready.');
+          resolve();
+        });
+      });
+
+    // Otherwise we need to prep protractor's selenium
+    } else {
+      const webdriverManagerPath = path.resolve(
+        'node_modules',
+        '.bin',
+        'webdriver-manager'
+      );
+
+      const results = spawn.sync(
+        webdriverManagerPath,
+        [
+          'update',
+          '--standalone', 'false',
+          '--gecko', 'false'
+        ],
+        spawnOptions
+      );
+
+      if (results.error) {
+        reject(results.error);
+        return;
+      }
+
+      logger.info('Selenium server is ready.');
+      resolve();
+    }
+  });
+}
+
+/**
+ * Spawns the build process.  Captures the config used.
+ */
+function spawnBuild(argv, skyPagesConfig, webpack) {
+  return new Promise((resolve, reject) => {
+
+    if (argv.build === false) {
+      logger.info('Skipping build step');
+
+      const file = 'dist/metadata.json';
+      if (!fs.existsSync(file)) {
+        logger.info(`Unable to skip build step.  "${file}" not found.`);
+      } else {
+        return resolve({
+          metadata: fs.readJsonSync(file)
+        });
+      }
+    }
+
+    logger.info('Running build...');
+    build(argv, skyPagesConfig, webpack)
+      .then(stats => {
+        logger.info('Build complete.');
+        resolve(stats.toJson().chunks);
+      })
+      .catch(reject);
+  });
+}
+
+/**
+ * Spawns the necessary commands for e2e and visual tests.
+ * Assumes build was ran.
+ * @name start
+ */
+function start(command, argv, skyPagesConfig, webpack) {
+  startTime = new Date().getTime();
+  process.on('SIGINT', killServers);
+
+  let specsPath;
+  if (command === 'e2e') {
```

Comment:
Since you're already splitting this out of the actual `e2e` and `visual` cli files, what do you think about `specsPath` just being an argument?

Reply:
It'd be yet another parameter to pass into this method, but I understand why that's better. I'll work on it...

Reply:
@Blackbaud-BobbyEarl I refactored this to check for the specs pattern found in the Protractor config files.